### PR TITLE
Timer triggered functions

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Implements part of #218

### CODE REVIEW

File: examples/timer/index.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import * as azure from '@pulumi/azure';
+
+const resourceGroup = new azure.core.ResourceGroup('example', { location: 'West US 2' });
+
+new azure.appservice.TimerSubscription('everyminute', {
+    resourceGroup,
+    schedule: '0 * * * * *',
```
Comment (CyrusNajmabadi) [inline examples/timer/index.ts]: in awsx we also allowed a richer data structure for schedules: https://github.com/pulumi/pulumi-awsx/blob/master/nodejs/awsx/autoscaling/schedule.ts

Can we also do something like that here?

File: examples/timer/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as azure from '@pulumi/azure';
+
+const resourceGroup = new azure.core.ResourceGroup('example', { location: 'West US 2' });
+
+new azure.appservice.TimerSubscription('everyminute', {
+    resourceGroup,
+    schedule: { second: 0 },
```
Comment (CyrusNajmabadi) [inline examples/timer/index.ts]: i'm not certain how `second: 0` translates to `everyminute`.

File: sdk/nodejs/appservice/zMixins_timer.ts [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    function checkRange(
+            val: number | undefined, name: keyof ScheduleArgs,
+            minInclusive: number, maxInclusive: number) {
+        if (val !== undefined) {
+            if (val < minInclusive || val > maxInclusive) {
+                throw new Error(`Value for [args.${name}] was not in the inclusive range [${minInclusive}, ${maxInclusive}].`);
+            }
+        }
+    }
+}
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/appservice/zMixins_timer.ts]: pity we can't share all this code.

File: examples/timer/index.ts (hunk 2 above)
Comment (mikhailshilkov) [reply examples/timer/index.ts]: It translated to "0 * * * * *" which means "every time second = 0" means every minute. Not super obvious, like most cron expressions...

Comment (CyrusNajmabadi) [reply examples/timer/index.ts]: for the purposes of an example how about `{ minute: 5 }` and call it `everyfiveminutes` :)

Comment (mikhailshilkov) [reply examples/timer/index.ts]: `{ minute: 5}` will trigger when minute is 5, every second. So, 60 times per hour. Every five minutes is `0 */5 * * * *`.

Comment (CyrusNajmabadi) [reply examples/timer/index.ts]: ok.  can we do `{ second: 0, minute: 5 }` :D

Comment (mikhailshilkov) [reply examples/timer/index.ts]: That is going to run once per hour, at X:05:00

**Comment by mikhailshilkov** [comment]
@CyrusNajmabadi @lukehoban Anything preventing the merge of this PR?

**Comment by CyrusNajmabadi** [comment]
Had to fix a merge conflict.  Once this passes, i'll merge in.  If i forget def ping me!

**Pull request merged**